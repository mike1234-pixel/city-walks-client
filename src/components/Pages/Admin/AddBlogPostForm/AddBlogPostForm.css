/** @format */
.add-blog-post-form {
  padding-top: 20px;
  padding-bottom: 20px;
}

/*# sourceMappingURL=AddBlogPostForm.css.map */
