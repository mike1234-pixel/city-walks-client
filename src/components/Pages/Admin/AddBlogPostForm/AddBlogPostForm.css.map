{"version":3,"sourceRoot":"","sources":["AddBlogPostForm.scss"],"names":[],"mappings":"AAAA;AAEA;EACE;EACA","file":"AddBlogPostForm.css"}