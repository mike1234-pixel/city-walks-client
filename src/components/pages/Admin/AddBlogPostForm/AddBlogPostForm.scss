/** @format */

.add-blog-post-form {
  padding: 20px;
}
