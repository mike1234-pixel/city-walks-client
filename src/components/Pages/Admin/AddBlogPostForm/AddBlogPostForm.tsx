import React, { useState } from "react"
import { MDBInput, MDBBtn, MDBIcon } from "mdbreact"
import axios, { AxiosError, AxiosResponse } from "axios";
import qs from "qs";
import toTitleCase from '../../../../functions/toTitleCase'
import BlogPost from "../../../../types/PostRequests/BlogPost";
import './AddBlogPostForm.scss'

const AddBlogPostForm: React.FC = () => {

  const [postTitle, setPostTitle] = useState<string>("")
  const [postSubTitle, setPostSubTitle] = useState<string>("")
  const [postContent, setPostContent] = useState<string>("")
  const [postImgLink, setPostImgLink] = useState<string>("")
  const [postAuthor, setPostAuthor] = useState<string>("")

  const handleChange: (event: React.ChangeEvent<any>) => void = (event) => {
    switch (event.target.name) {
      case "post-title":
        setPostTitle(event.target.value)
        break;
      case "post-subtitle":
        setPostSubTitle(event.target.value)
        break;
      case "post-content":
        setPostContent(event.target.value)
        break;
      case "post-img-link":
        setPostImgLink(event.target.value)
        break;
      case "post-author":
        setPostAuthor(event.target.value)
        break;
    }
  }

  const handleSubmit: (event: React.FormEvent) => void = (event) => {
    event.preventDefault()

    let payload: BlogPost = {
      title: toTitleCase(postTitle),
      subtitle: toTitleCase(postSubTitle),
      content: postContent,
      img: postImgLink,
      author: toTitleCase(postAuthor)
    };

    axios
      .post("https://city-walks.herokuapp.com/add-blog-post", qs.stringify(payload))
      .then((res: AxiosResponse) => {
        if (res.data === "blog post saved") {
          console.log("blog post saved.")
        } else {
          console.log("blog not saved")
        }
      }).catch((err: AxiosError) => {
        console.log(err)
      });;

    alert("Blog Post Submitted")
    setPostTitle("")
    setPostSubTitle("")
    setPostContent("")
    setPostImgLink("")
    setPostAuthor("")
    window.scrollTo(0, 0);
  }

  return (
    <div>
      <h2>Add Blog Post</h2>
      <form onSubmit={handleSubmit} className="add-blog-post-form">
        <MDBInput type="text" name="post-title" id="post-title" value={postTitle} label="post title" onChange={handleChange} required />
        <MDBInput type="text" name="post-subtitle" id="post-subtitle" value={postSubTitle} label="post sub-title" onChange={handleChange} required />
        <MDBInput type="textarea" rows="10" name="post-content" id="post-content" value={postContent} label="post content" onChange={handleChange} required />
        <MDBInput type="text" name="post-img-link" id="post-img-link" value={postImgLink} label="post image link" onChange={handleChange} required />
        <MDBInput type="text" name="post-author" id="post-author" value={postAuthor} label="post author" onChange={handleChange} required />
        <MDBBtn outline color="elegant" type="submit">
          Add Blog Post <MDBIcon icon="plus" />
        </MDBBtn>
      </form>
    </div>
  )
}

export default AddBlogPostForm