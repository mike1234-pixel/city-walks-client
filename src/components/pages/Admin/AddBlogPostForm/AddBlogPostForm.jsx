import { useState } from "react"
import { MDBInput, MDBBtn, MDBIcon } from "mdbreact"
import axios from "axios";
import qs from "qs";
import toTitleCase from '../../../../functions/toTitleCase'
import './AddBlogPostForm.css'

const AddBlogPostForm = () => {

    const [postTitle, setPostTitle] = useState("")
    const [postSubTitle, setPostSubTitle] = useState("")
    const [postContent, setPostContent] = useState("")
    const [postImgLink, setPostImgLink] = useState("")
    const [postAuthor, setPostAuthor] = useState("")

    const handleChange = (event) => {
        switch(event.target.name) {
            case "post-title":
              setPostTitle(event.target.value)
              break;
            case "post-subtitle":
              setPostSubTitle(event.target.value)
              break;
            case "post-content":
              setPostContent(event.target.value)
              break;
            case "post-img-link":
              setPostImgLink(event.target.value)
              break;
            case "post-author":
              setPostAuthor(event.target.value)
              break;
          } 
    }

    const handleSubmit = (event) => {
        event.preventDefault()

        let payload = {
            title: toTitleCase(postTitle),
            subtitle: toTitleCase(postSubTitle),
            content: postContent,
            img: postImgLink,
            author: toTitleCase(postAuthor)
          };

        axios
        .post("http://localhost:5000/add-blog-post", qs.stringify(payload))
        .then((res, err) => {
          if (err) {
            console.log(err);
          } else if (res.data === "blog post saved") {
            console.log("blog post saved.")
          } else {
            console.log("blog not saved")
          }
        });

        alert("Blog Post Submitted")
        setPostTitle("")
        setPostSubTitle("")
        setPostContent("")
        setPostImgLink("")
        setPostAuthor("")
        window.scrollTo(0, 0);
    }

    return (
    <div>
        <h2>Add Blog Post</h2>
        <form onSubmit={handleSubmit} className="add-blog-post-form">
            <MDBInput type="text" name="post-title" id="post-title" value={postTitle} label="post title" onChange={handleChange} required/>
            <MDBInput type="text" name="post-subtitle" id="post-subtitle" value={postSubTitle} label="post sub-title" onChange={handleChange} required/>
            <MDBInput type="textarea" rows="10" name="post-content" id="post-content" value={postContent} label="post content" onChange={handleChange} required/>
            <MDBInput type="text" name="post-img-link" id="post-img-link" value={postImgLink} label="post image link" onChange={handleChange} required/>
            <MDBInput type="text" name="post-author" id="post-author" value={postAuthor} label="post author" onChange={handleChange} required/>
            <MDBBtn outline color="elegant" type="submit">
                Add Blog Post <MDBIcon icon="plus" />
          </MDBBtn>
        </form>
    </div>
    )
}

export default AddBlogPostForm