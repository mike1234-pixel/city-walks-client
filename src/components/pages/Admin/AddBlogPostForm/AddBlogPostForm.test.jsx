import renderer from 'react-test-renderer'
import AddBlogPostForm from "./AddBlogPostForm"

// snapshot test
it('AddBlogPostForm Page renders correctly', () => {  

  const tree = renderer
    .create(
        <AddBlogPostForm />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});