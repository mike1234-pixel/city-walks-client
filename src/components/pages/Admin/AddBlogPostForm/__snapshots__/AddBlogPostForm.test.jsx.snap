// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddBlogPostForm Page renders correctly 1`] = `
<div>
  <h2>
    Add Blog Post
  </h2>
  <form
    className="add-blog-post-form"
    onSubmit={[Function]}
  >
    <div
      className="md-form"
    >
      
      <input
        aria-disabled={false}
        className="form-control"
        data-test="input"
        disabled={false}
        id="post-title"
        name="post-title"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onInput={[Function]}
        required={true}
        type="text"
        value=""
      />
      <label
        aria-labelledby=""
        className=""
        data-error=""
        data-success=""
        htmlFor="post-title"
        id=""
        onClick={[Function]}
      >
        post title
      </label>
    </div>
    <div
      className="md-form"
    >
      
      <textarea
        aria-disabled={false}
        className="md-textarea form-control"
        data-test="input"
        disabled={false}
        id="post-content"
        name="post-content"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onInput={[Function]}
        required={true}
        rows="10"
        value=""
      />
      <label
        aria-labelledby=""
        className=""
        data-error=""
        data-success=""
        htmlFor="post-content"
        id=""
        onClick={[Function]}
      >
        post content
      </label>
    </div>
    <div
      className="md-form"
    >
      
      <input
        aria-disabled={false}
        className="form-control"
        data-test="input"
        disabled={false}
        id="post-img-link"
        name="post-img-link"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onInput={[Function]}
        required={true}
        type="text"
        value=""
      />
      <label
        aria-labelledby=""
        className=""
        data-error=""
        data-success=""
        htmlFor="post-img-link"
        id=""
        onClick={[Function]}
      >
        post image link
      </label>
    </div>
    <button
      className="btn-elegant btn-outline-elegant btn Ripple-parent"
      data-test="button"
      onMouseUp={[Function]}
      onTouchStart={[Function]}
      type="submit"
    >
      Add Blog Post 
      <i
        className="fa fa-plus"
        data-test="fa"
      />
      <div
        className="Ripple Ripple-outline "
        data-test="waves"
        style={
          Object {
            "height": "0px",
            "left": "0px",
            "top": "0px",
            "width": "0px",
          }
        }
      />
    </button>
  </form>
</div>
`;
